frontend/src/components/FilterDropdown.tsx

```diff
@@ -0,0 +1,197 @@
+import { useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+
+import { FilterParams } from "@/api/units";
+import { AvailabilityDropDown } from "@/components/AvailabilityDropDown";
+import { BedBathDropDown } from "@/components/BedBathDropDown";
+import { FilterText } from "@/components/FilterCommon";
+import { PriceDropDown } from "@/components/PriceDropDown";
+import { SortDropDownComp } from "@/components/SortDropDown";
+
+const AllFiltersContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: flex-start;
+  margin-left: 95px;
+  margin-right: 95px;
+  margin-top: 95px;
+  gap: 16px;
+`;
+
+const FiltersFirstRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: center;
+  gap: 28px;
+  flex-wrap: wrap;
+`;
+
+const SearchBarInput = styled.input`
+  padding: 3px;
+  min-width: 16rem;
+  border: 0;
+
+  &::placeholder {
+    color: #000;
+    font-family: Montserrat;
+    font-size: 16px;
+    font-weight: 600;
+  }
+
+  &:focus {
+    outline: none;
+  }
+`;
+
+const SearchIcon = styled.img`
+  height: 20px;
+  width: 20px;
+`;
+
+const SearchBarContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: center;
+  padding: 10px;
+  background-color: #fff;
+  border-radius: 5px;
+  border: 0.5px solid #cdcaca;
+  box-shadow: 1px 1px 2px 0px rgba(188, 186, 183, 0.4);
+`;
+
+const ResetIcon = styled.img`
+  height: 25px;
+  width: 25px;
+`;
+
+const ResetFilterButton = styled.button`
+  background-color: transparent;
+  border-color: transparent;
+`;
+
+const ResetFilterText = styled(FilterText)`
+  color: #b64201;
+  font-weight: 500;
+  padding-top: 2px;
+`;
+
+const ResetFilterRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: flex-center;
+  gap: 8px;
+`;
+
+type FilterDropdownProps = {
+  refreshUnits(filterParams: FilterParams): void;
+};
+
+export const FilterDropdown = (props: FilterDropdownProps) => {
+  const [numBedBath, setNumBedBath] = useState({ beds: 1, baths: 0.5 });
+  const [searchText, setSearchText] = useState("");
+  const [sortIndex, setSortIndex] = useState(0);
+  const [availabilityIndex, setAvailabilityIndex] = useState(0);
+  const [price, setPrice] = useState({ minPrice: -1, maxPrice: -1 });
+
+  const resetFunctions = useRef(Array<() => void>);
```

Comment:
Interesting, I've never seen a `useRef` used this way. It seems to do the job though 🤷🏼 . 

One suggestion I have that might be a better way to architect this is to keep the state of the dropdown in the `FilterDropdown` component and just pass the value and setValue function of the state down to the dropdowns. That way `FilterDropdown` does not need to call functions in the dropdown components to reset the filters. Essentially the individual dropdown components will be completely stateless.

For example, the `AvailabilityDropDown` would take these props:
```typescript
const [availabilityIndex, setAvailabilityIndex] = useState(0);
...
return (
...
<AvailabilityDropDown value={availabilityIndex} setValue={setAvailabilityIndex} />
...
)
```

Internally, `AvailabilityDropDown` will call `setAvailabilityIndex` when the user updates their selection. When resetting filters, `FilterDropdown` will just call `setAvailabilityIndex` to reset to the default value. This should clean up the logic and eliminate the need to pass functions up to  `FilterDropdown` (the parent function).

Let me know if that makes sense.

Reply:
In general, React encourages a unidirectional data flow paradigm where data flows from parent nodes to child nodes (not child -> parent). The current implementation is kind of a bidirectional flow since you are passing a callback function from `AvailabilityDropDown` (the child) to `FilterDropdown` (the parent).

Here's a short article explaining that concept: https://www.educative.io/answers/what-is-unidirectional-data-flow-in-react